lib/utils.rb

```diff
@@ -31,5 +32,31 @@ def Utils.paramhash_to_str params
   params.collect {|k, v| if (v == true) then ";#{k}" else ";#{k}=#{v}" end}.join("")
 end
 
+def Utils.check_route_matches_proxy route, proxy
+  regexp = /<sip:(.*?)@(.*?):(.*?);(.*?)>/i
+  proxy_address = proxy.instance_variable_get(:@ip)
```

Comment:
(Also, mildly sad that - presumably - `proxy.ip` is not an IP address. Worth a comment and I ought to fix that up.)